lib/elements/satin_column.py

```diff
@@ -1066,12 +1073,34 @@ def do_e_stitch(self):
             self.random_width_decrease.any() or self.random_width_increase.any() or self.random_zigzag_spacing,
         )
 
+        short_pairs = self.inset_short_stitches_sawtooth(pairs)
+        max_stitch_length = self.max_stitch_length_px
+        length_sigma = self.random_split_jitter
+        random_phase = self.random_split_phase
+        min_split_length = self.min_random_split_length_px
+        seed = self.random_seed
+        last_point = None
         # "left" and "right" here are kind of arbitrary designations meaning
         # a point from the first and second rail respectively
-        for left, right in pairs:
-            patch.add_stitch(left)
-            patch.add_stitch(right)
-            patch.add_stitch(left)
+        for i, (left, right), (a_short, b_short) in zip(itertools.count(0), pairs, short_pairs):
+            check_stop_flag()
+            split_points, _ = self.get_split_points(
+                left, right, a_short, b_short, max_stitch_length,
+                None, length_sigma, random_phase, min_split_length,
+                prng.join_args(seed, 'satin-split', 2 * i + 1))
+
+            # zigzag spacing is wider than stitch length, subdivide
+            if last_point is not None and max_stitch_length is not None and self.zigzag_spacing > max_stitch_length:
+                points, _ = self.get_split_points(last_point, left, last_point, left, max_stitch_length)
+                patch.add_stitches(points)
+
+            patch.add_stitch(a_short, ("edge"))
+            patch.add_stitches(split_points, ("split_stitch"))
+            patch.add_stitch(b_short, ("edge"))
+            patch.add_stitches(split_points[::-1], ("split_stitch"))
+            patch.add_stitch(a_short, ("edge"))
```

Comment:
We should be using `left` instead of `a_short` to keep the line straight.

Reply:
Thank you for your review. I was thinking about this. But I actually see this as a feature. There may are artistic use cases where you do not want this to be a straight line. If you need a straight line, enter two values and it is possible too. I really do like to have all the options.